shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -316,6 +316,20 @@ public void removeEngineLifecycleListener(@NonNull EngineLifecycleListener liste
     engineLifecycleListeners.remove(listener);
   }
 
+  /**
+   * Notify the Dart VM that the system is experiencing memory pressure, and
+   * extra work should be taken to clean up internal resources and collect
+   * garbage.
+   *
+   * This does not notify a Flutter application about memory pressure. For that,
+   * use the {@link SystemChannel#sendMemoryPressureWarning}.
+   */
+  public void notifyLowMemoryWarning() {
```

Comment:
for modularity, put this in the DartExecutor since it's a dart specific thing

Reply:
Done